apis/python/tests/test_point_cloud.py

```diff
@@ -0,0 +1,705 @@
+            "x": [10, 20, 30, 40, 50],
+            "y": [4.1, 5.2, 6.3, 7.4, 8.5],
+            "z": [0, 1, 2, 3, 4],
+        }
+        rb = pa.Table.from_pydict(pydict)
+        ptc.write(rb)
+
+    with soma.PointCloud.open(uri, "r") as ptc:
+        with pytest.raises(exc_type):
+            ptc.read_spatial_region(region=region)
+
+
+def point_cloud_read_spatial_region_transform_setup(uri, transform, input_axes, kwargs):
+    schema = pa.schema([("x", pa.float64()), ("y", pa.float64())])
+
+    with soma.PointCloud.create(uri, schema=schema) as ptc:
+        pydict = {
+            "soma_joinid": [1, 2, 3, 4, 5],
+            "x": [10, 20, 30, 40, 50],
+            "y": [4.1, 5.2, 6.3, 7.4, 8.5],
+        }
+        rb = pa.Table.from_pydict(pydict)
+        ptc.write(rb)
+
+    output_names = ("x", "y")
+    input_names = tuple(axis.name for axis in input_axes)
+
+    with soma.PointCloud.open(uri, "r") as ptc:
+        read_spatial_region = ptc.read_spatial_region(
+            region_transform=transform(
+                input_axes=input_names, output_axes=output_names, **kwargs
+            )
+        )
+
+        assert read_spatial_region.data.concat() == ptc.read().concat()
+
+        data_coordinate_space = read_spatial_region.data_coordinate_space
+        assert (
+            data_coordinate_space.axis_names
+            == ptc.coordinate_space.axis_names
+            == output_names
+        )
+        assert len(data_coordinate_space) == len(output_names)
+
+        output_coordinate_space = read_spatial_region.output_coordinate_space
+        assert output_coordinate_space.axes == input_axes
+        assert output_coordinate_space.axis_names == input_names
+        assert len(output_coordinate_space) == len(input_names)
+
+        assert isinstance(read_spatial_region.coordinate_transform, transform)
+
+    return read_spatial_region.coordinate_transform
+
+
+def test_point_cloud_read_spatial_region_identity_transform(tmp_path):
+    coordinate_transform = point_cloud_read_spatial_region_transform_setup(
+        uri=tmp_path.as_uri(),
+        transform=soma.IdentityTransform,
+        input_axes=(soma.Axis(name="x"), soma.Axis(name="y")),
+        kwargs={},
+    )
+
+    # Identity will always have a scale of 1.0
+    assert list(coordinate_transform.scale_factors) == [1, 1]
+    assert coordinate_transform.scale == 1.0
+    assert list(coordinate_transform.inverse_transform().scale_factors) == [1, 1]
+
+
+@pytest.mark.parametrize(
+    "name, scale, expected_scale_factors, expected_scale",
+    [
+        ("Identity", 1, [1, 1], 1.0),
+        ("Scale up", 2, [0.5, 0.5], 0.5),
+        ("Negative scale up", -2, [-0.5, -0.5], -0.5),
+        ("Very large", 1e6, [1e-6, 1e-6], 1e-6),
+        ("Very small", 1e-6, [1e6, 1e6], 1e6),
+        ("Scale down", 0.5, [2, 2], 2.0),
+        ("Negative scale down", -0.5, [-2, -2], -2.0),
+        ("Zero scale", 0, [float("inf"), float("inf")], float("inf")),
```

Comment:
This throws a `ZeroDivisionError`. That feels right to me. We can just break it out into a test that checks that error is thrown.